crates/store/src/db/models/conv.rs

```diff
@@ -131,6 +131,33 @@ impl SqlTypeConvert for NoteTag {
     }
 }
 
+impl SqlTypeConvert for StorageSlotType {
+    type Raw = i32;
+
+    #[inline(always)]
+    fn from_raw_sql(raw: Self::Raw) -> Result<Self, DatabaseTypeConversionError> {
+        #[derive(Debug, thiserror::Error)]
+        #[error("invalid storage slot type value {0}")]
+        struct ValueError(i32);
+
+        Ok(match raw {
+            0 => StorageSlotType::Map,
+            1 => StorageSlotType::Value,
+            invalid => {
+                return Err(Self::map_err(ValueError(invalid)));
+            },
+        })
```

Comment:
This is using inverse encoding of what we use in `miden-base` - i.e., there `0` means value slot and `1` means map slot. Is this intentional?

I think while this serialization is not directly related to `miden-base`, it would probably still be a good idea to keep the same approach (i.e., `0` for value and `1` for map).